pandas/tests/tools/test_to_datetime.py

```diff
@@ -135,6 +135,17 @@ def test_to_datetime_format_YYYYMMDD_with_nat(self, cache):
         result = to_datetime(ser2, format="%Y%m%d", cache=cache)
         tm.assert_series_equal(result, expected)
 
+    def test_to_datetime_format_YYYYMM_with_nat(self, cache):
+        # https://github.com/pandas-dev/pandas/issues/50237
+        ser = Series([198012, 198012] + [198101] * 5)
+        expected = Series(
+            [Timestamp("19801201"), Timestamp("19801201")] + [Timestamp("19810101")] * 5
+        )
+        expected[2] = np.nan
```

Comment:
If `np.nan` was `pd.datetime64('NaT')` would this hit the `elif` block added in this PR  (and if so could you add a test for it)?

Reply:
hey - no, that would hit this branch:

https://github.com/pandas-dev/pandas/blob/e8e77c781e68d05896e852007d17c12112931a33/pandas/_libs/tslibs/strptime.pyx#L163-L165

, which is covered by a bunch of tests, e.g.

https://github.com/pandas-dev/pandas/blob/e8e77c781e68d05896e852007d17c12112931a33/pandas/tests/tools/test_to_datetime.py#L168-L189